packages/core/src/storage.ts

```diff
@@ -125,25 +129,47 @@ export function create({ token, host }: StorageConfig): StorageAPI {
         },
       });
       if (res.ok) {
-        const json = await res.json();
-        return json;
+        const { status_code, ...rest }: RequestAlias = await res.json();
+        const request: StorageRequest = {
+          ...rest,
+          status: toSnakeCase(
+            status_code
+          ).toUpperCase() as StorageRequestStatus,
+        };
+        return request;
       }
       const err = await res.text();
       throw new Error(err);
     },
-    status: async function status(id: string): Promise<RequestInfo> {
-      const res = await fetch(`${host}/storagerequest/${id}`, {
-        method: "GET",
-        headers: {
-          Authorization: `Bearer ${token}`,
-        },
-      });
-      if (res.ok) {
-        const json = await res.json();
-        return json;
-      }
-      const err = await res.text();
-      throw new Error(err);
+    status: async function status(id: string): Promise<StatusRequest> {
```

Comment:
much simpler logic here, and the core parsing is pulled out into the function below